Source/Core/Common/Config/ConfigInfo.h

```diff
@@ -30,10 +34,30 @@ struct Location
 };
 
 template <typename T>
-struct Info
+struct CachedValue
 {
-  Info(const Location& location_, const T& default_value_)
-      : location{location_}, default_value{default_value_}
+  T value;
+  u64 config_version;
+};
+
+template <typename T>
+class Info
+{
+public:
+  constexpr Info(const Location& location, const T& default_value)
+      : m_location{location}, m_default_value{default_value}, m_cached_value{default_value, 0}
+  {
+  }
+
+  Info(const Info<T>& other)
```

Comment:
This can be implemented using the copy assignment operator to avoid duplicating code:

```cpp
Info(const Info<T>& other) { *this = other; }
```

(or with the copy-and-swap idiom, but I prefer this one for symmetry with the move constructor and for simplicity reasons)